packages/contracts-bedrock/src/L2/Callbacks.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { Identifier, IL2ToL2CrossDomainMessenger } from "interfaces/L2/IL2ToL2CrossDomainMessenger.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+interface ICallbackGreeterMetadata {
+    /// @notice Callback context to be executed (along with params if any) once the callback is received
+    /// @param to Target address
+    /// @param selector Function selector
+    struct CallbackContext {
+        address to;
+        bytes4 selector;
+    }
+
+    /// @notice Emitted when the greeting is set
+    /// @param greeting New greeting
+    event GreetingSet(string greeting);
+
+    /// @notice Emitted when an async call is sent
+    /// @param contextNonce Nonce of the context
+    event AsyncCallSent(uint224 contextNonce);
+
+    /// @notice Emitted when the remote greeter is set
+    /// @param remoteGreeter Address of the remote greeter
+    event RemoteGreeterSet(address remoteGreeter);
+
+    /// @notice Emitted when the callback is executed
+    /// @param contextNonce Nonce of the context
+    event CallbackExecuted(uint224 contextNonce);
+
+    /// @notice Thrown when caller is not the owner
+    error OnlyOwner();
+
+    /// @notice Thrown when the callback fails
+    error CallbackFailed();
+}
+
+/**
+ * @title CallbackGreeter
+ * @notice Initiates a remote greeting call to a greeter contract on a different chain via the CallbackEntrypoint. It
+ * also handles the callback received from the remote greeter.
+ */
+contract CallbackGreeter is ICallbackGreeterMetadata {
+    /// @notice Address of the L2ToL2CrossDomainMessenger contract
+    address public constant L2_TO_L2_CROSS_DOMAIN_MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
```

Comment:
Looks like there is not much consistency in the usage or predeploys addresses. I would like to use the SuperTokenBridge one, where this constant is called simply `MESSENGER`

https://github.com/ethereum-optimism/optimism/blob/0d0f69d59cfbce954710636efdf9812d9a38fead/packages/contracts-bedrock/src/L2/SuperchainTokenBridge.sol#L46

https://github.com/ethereum-optimism/optimism/blob/0d0f69d59cfbce954710636efdf9812d9a38fead/packages/contracts-bedrock/src/L2/SuperchainWETH.sol#L151